pkg/operator/target_config_reconciler.go

```diff
@@ -0,0 +1,627 @@
+		ownerReference,
+	}
+	controller.EnsureOwnerRef(required, ownerReference)
+
+	for i, _ := range required.Webhooks {
+		if required.Webhooks[i].ClientConfig.Service != nil {
+			required.Webhooks[i].ClientConfig.Service.Namespace = c.namespace
+		}
+	}
+
+	return resourceapply.ApplyMutatingWebhookConfigurationImproved(c.ctx, c.kubeClient.AdmissionregistrationV1(), c.eventRecorder, required, resourceapply.NewResourceCache())
+}
+
+func (c *TargetConfigReconciler) manageValidatingWebhook(leaderWorkerSetOperator *leaderworkersetapiv1.LeaderWorkerSetOperator) (*admissionv1.ValidatingWebhookConfiguration, bool, error) {
+	required := resourceread.ReadValidatingWebhookConfigurationV1OrDie(bindata.MustAsset("assets/lws-operator/validatingwebhook.yaml"))
+	ownerReference := metav1.OwnerReference{
+		APIVersion: "operator.openshift.io/v1",
+		Kind:       "LeaderWorkerSetOperator",
+		Name:       leaderWorkerSetOperator.Name,
+		UID:        leaderWorkerSetOperator.UID,
+	}
+	required.OwnerReferences = []metav1.OwnerReference{
+		ownerReference,
+	}
+	controller.EnsureOwnerRef(required, ownerReference)
+
+	for i, _ := range required.Webhooks {
+		if required.Webhooks[i].ClientConfig.Service != nil {
+			required.Webhooks[i].ClientConfig.Service.Namespace = c.namespace
+		}
+	}
+
+	return resourceapply.ApplyValidatingWebhookConfigurationImproved(c.ctx, c.kubeClient.AdmissionregistrationV1(), c.eventRecorder, required, resourceapply.NewResourceCache())
+}
+
+func (c *TargetConfigReconciler) manageServiceMonitor(leaderWorkerSetOperator *leaderworkersetapiv1.LeaderWorkerSetOperator) (bool, error) {
+	required := resourceread.ReadUnstructuredOrDie(bindata.MustAsset("assets/lws-operator/servicemonitor.yaml"))
+	required.SetNamespace(c.namespace)
+	_, changed, err := resourceapply.ApplyKnownUnstructured(c.ctx, c.dynamicClient, c.eventRecorder, required)
+	return changed, err
+}
+
+func (c *TargetConfigReconciler) manageDeployments(leaderWorkerSetOperator *leaderworkersetapiv1.LeaderWorkerSetOperator) (*appsv1.Deployment, bool, error) {
+	required := resourceread.ReadDeploymentV1OrDie(bindata.MustAsset("assets/lws-operator/deployment.yaml"))
+	required.Namespace = c.namespace
+	ownerReference := metav1.OwnerReference{
+		APIVersion: "operator.openshift.io/v1",
+		Kind:       "LeaderWorkerSetOperator",
+		Name:       leaderWorkerSetOperator.Name,
+		UID:        leaderWorkerSetOperator.UID,
+	}
+	required.OwnerReferences = []metav1.OwnerReference{
+		ownerReference,
+	}
+	controller.EnsureOwnerRef(required, ownerReference)
+
+	if c.targetImage != "" {
+		images := map[string]string{
+			"${IMAGE}": c.targetImage,
+		}
+
+		for i := range required.Spec.Template.Spec.Containers {
+			for env, img := range images {
+				if required.Spec.Template.Spec.Containers[i].Image == env {
+					required.Spec.Template.Spec.Containers[i].Image = img
+					break
+				}
+			}
+		}
+	}
+
+	switch leaderWorkerSetOperator.Spec.LogLevel {
+	case operatorv1.Normal:
```

Comment:
do the zap levels work the same as the kube ones? I have opted for info x debug for now :D

Reply:
I did this arbitrarily :). We can adjust later.